pkg/openevec/adam.go

```diff
@@ -25,3 +31,111 @@ func (openEVEC *OpenEVEC) AdamStart() error {
+		return fmt.Errorf("failed to reencrypt existing configs: %w", err)
+	}
+
+	if err = changer.setControllerAndDev(ctrl, dev); err != nil {
+		return fmt.Errorf("setControllerAndDev: %w", err)
+	}
+
+	err = ctrl.UploadSigningCert(newSignCert)
+	if err != nil {
+		log.Errorf("cannot upload signing cert: %s", err.Error())
+	} else {
+		log.Infof("Signing cert uploaded")
+	}
+	return nil
+}
+
+func reencryptConfigs(ctrl controller.Cloud, dev *device.Ctx, newSignCert []byte) error {
+	// get device certificate from the controller
+	devCert, err := ctrl.DevCertsGet(dev)
+	if err != nil {
+		return fmt.Errorf("cannot get device certificate from cloud: %w", err)
+	}
+
+	// get signing certificate from the controller
+	oldSignCert, err := ctrl.SigningCertGet()
+	if err != nil {
+		log.Error("cannot get cloud's signing certificate. will use plaintext")
+		return nil
+	}
+
+	edenHome, err := utils.DefaultEdenDir()
+	if err != nil {
+		return fmt.Errorf("DefaultEdenDir: %w", err)
+	}
+	keyPath := filepath.Join(edenHome, defaults.DefaultCertsDist, "signing-key.pem")
+	ctrlPrivKey, err := os.ReadFile(keyPath)
+	if err != nil {
+		return fmt.Errorf("cannot read %s: %w", keyPath, err)
+	}
+
+	oldCryptoConfig, err := utils.GetCommonCryptoConfig(devCert, oldSignCert, ctrlPrivKey)
+	if err != nil {
+		return fmt.Errorf("GetCommonCryptoConfig: %w", err)
+	}
+
+	newCryptoConfig, err := utils.GetCommonCryptoConfig(devCert, newSignCert, ctrlPrivKey)
+	if err != nil {
+		return fmt.Errorf("GetCommonCryptoConfig: %w", err)
+	}
+
+	cipherCtx, err := utils.CreateCipherCtx(newCryptoConfig)
+	if err != nil {
+		return fmt.Errorf("CreateCipherCtx: %w", err)
+	}
+	utils.AddCipherCtxToDev(dev, cipherCtx)
+
+	// re-encrypt all app configs with the new signing certificate
+	appConfigs := ctrl.ListApplicationInstanceConfig()
+	for _, config := range appConfigs {
+		if config.CipherData != nil {
+			encBlock, err := utils.CryptoConfigUnwrapper(config.CipherData, oldCryptoConfig)
+			if err != nil {
+				return fmt.Errorf("CryptoConfigUnwrapper error: %w", err)
+			}
+
+			cipherData, err := utils.CryptoConfigWrapper(encBlock, newCryptoConfig, cipherCtx)
+			if err != nil {
+				return fmt.Errorf("CryptoConfigWrapper error: %w", err)
+			}
+
+			// set cipher data encrypted with the new signing certificate
+			config.CipherData = cipherData
+		}
+	}
+
+	// re-encrypt all datastore configs with the new signing certificate
+	dsConfigs := ctrl.ListDataStore()
+	for _, config := range dsConfigs {
+		if config.CipherData != nil {
+			encBlock, err := utils.CryptoConfigUnwrapper(config.CipherData, oldCryptoConfig)
+			if err != nil {
+				return fmt.Errorf("CryptoConfigUnwrapper error: %w", err)
+			}
+
+			cipherData, err := utils.CryptoConfigWrapper(encBlock, newCryptoConfig, cipherCtx)
+			if err != nil {
+				return fmt.Errorf("CryptoConfigWrapper error: %w", err)
+			}
+
+			// set cipher data encrypted with the new signing certificate
+			config.CipherData = cipherData
+		}
+	}
+
```

Comment:
We also use cipher block here: https://github.com/lf-edge/eve-api/blob/main/proto/config/netconfig.proto#L171

Reply:
I noticed that we also use `CipherBlock` here, however this is not the version of EVE API that Eden uses. Why does Eden still use github.com/lf-edge/eve/api/go instead of github.com/lf-edge/eve-api/go?

Reply:
Fair question. We should update dependencies. But now we just rely on protobuf compatibility, so can use CipherBlock from [here](https://github.com/lf-edge/eve/blob/5a232aec4746e84cb9dc2b64f5d728c3926540df/api/go/config/netconfig.pb.go#L558). But it would be nice to support all of them (if it possible) in converter and write comment somewhere to not forget update in case of new cipherblocks.

Reply:
IMO we should just move to eve-api/go repository. We should use same versions as EVE (i.e. we use EVE 11.04 as default EVE for Eden repo, which uses eve-api/go, so we need to use the same API)

Reply:
I started with adam: https://github.com/lf-edge/adam/pull/121

Reply:
@milan-zededa thank you! I'm doing the update for Eden now

Reply:
@europaul I cherry-picked your commit to a [separate PR](https://github.com/lf-edge/eden/pull/935) so that we can merge it sooner (I need it to progress in my work). Hope it's OK